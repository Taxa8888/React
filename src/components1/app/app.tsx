import React, { ReactElement } from 'react';
import { Header } from '../header/header';
import { HeaderDescription } from '../headerdescription/headerdescription';
import { Sort } from '../sort/sort';
import { SortDescription } from '../sortdescription/sortdescription';
import { Main } from '../main/main';
import { Footer } from '../footer/footer';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { ErrorPage } from '../errorpage/errorpage';

export const App = (): ReactElement => {
    return (
        <Router>
            <Switch>
                <Route exact path={['/', '/search/*']}>
                    <Header />
                    <Sort />
                    <Main />
                </Route>
                <Route path="/movie/:id">
                    <HeaderDescription />
                    <SortDescription />
                    <Main />
                </Route>
                <Route path={['*']}>
                    <ErrorPage />
                </Route>
            </Switch>
            <Footer />
        </Router>
    );
};
