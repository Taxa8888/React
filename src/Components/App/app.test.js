import { shallow } from 'enzyme';
import React from 'react';
import { App } from './app';

/*global describe, it, expect*/

describe('App component', () => {
    it('should render App component', () => {
        const component = shallow(<App />);
        expect(component).toMatchSnapshot();
    });
});
