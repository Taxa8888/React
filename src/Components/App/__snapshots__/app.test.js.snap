// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App component 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      exact={true}
      path={
        Array [
          "/",
          "/search/*",
        ]
      }
    >
      <Header />
      <Sort />
      <Main />
    </Route>
    <Route
      path="/movie/:id"
    >
      <HeaderDescription />
      <SortDescription />
      <Main />
    </Route>
    <Route
      path={
        Array [
          "*",
        ]
      }
    >
      <ErrorPage />
    </Route>
  </Switch>
  <Footer />
</BrowserRouter>
`;
